src/main/surfrdUtilsMod.F90

```diff
@@ -145,6 +146,109 @@ subroutine convert_cft_to_pft( begg, endg, cftsize, wt_cft )
 
   end subroutine convert_cft_to_pft
 
+  !-----------------------------------------------------------------------
+  subroutine collapse_to_dom_landunits(pctwet, pctlak, pcturb, pcturb_tot, pctgla, begg, endg, n_dom_landunits, numurbl)
+    !
+    ! DESCRIPTION
+    ! Collapse to the top N dominant landunits (n_dom_landunits)
+    !
+    ! !USES:
+    use array_utils, only: find_k_max_indices
+    !
+    ! !ARGUMENTS:
+    ! Use begg and endg rather than 'bounds', because bounds may not be
+    ! available yet when this is called
+    integer, intent(in) :: begg  ! Beginning grid cell index
+    integer, intent(in) :: endg  ! Ending grid cell index
+    integer, intent(in) :: n_dom_landunits  ! # dominant landunits
+    integer, intent(in) :: numurbl  ! # urban landunits
+    ! These arrays modified in-place
+    real(r8), intent(inout) :: pctwet(begg:endg)
+    real(r8), intent(inout) :: pctlak(begg:endg)
+    real(r8), intent(inout) :: pcturb(begg:endg, numurbl)
+    real(r8), intent(inout) :: pcturb_tot(begg:endg)
+    real(r8), intent(inout) :: pctgla(begg:endg)
+    !
+    ! !LOCAL VARIABLES:
+    integer :: g  ! gridcell index
+    integer :: m  ! landunit index
+    integer :: n  ! index of the order of the dominant pfts
+    integer, allocatable :: max_indices(:)  ! array of dominant pft index values
+    integer, parameter :: landunits_lb = 1  ! lower bound of wt_landunits array
+    integer, parameter :: landunits_ub = 4  ! upper bound of wt_landunits array
+    real(r8) :: wt_landunits(landunits_ub)  ! 1: pctwet, 2: pctlak, 3: pcturb_tot, 4: pctgla
+    real(r8) :: wt_landunits_sum  ! sum of landunits weights
+    real(r8) :: wt_dom_landunit_sum  ! sum of dominant landunits wgts
+
+    character(len=*), parameter :: subname = 'collapse_to_dom_landunits'
+
+    !-----------------------------------------------------------------------
+
+    ! Find the top N dominant landunits to collapse the data to
+    ! n_dom_landunits < 0 is not allowed (error check in controlMod.F90)
+    ! Default value n_dom_landunits = 0 or a user-selected
+    !               n_dom_landunits = landunits_ub (EXISTS?) both mean
+    ! "do not collapse landunits" and skip over this subroutine's work
+    if (n_dom_landunits > 0 .and. n_dom_landunits < landunits_ub) then
+       allocate(max_indices(n_dom_landunits))
+       do g = begg, endg
+          max_indices = 0  ! initialize
+          wt_landunits(landunits_lb) = pctwet(g)
```

Comment:
Is it possible to change the logic in this subroutine so that the subroutine will work without modification if we add more landunits to the code?

Reply:
I agree with @billsacks it should be flexible to handle however many landunits are currently implemented in the code.

Reply:
Looking at options. My current favorite is to move the call collapse_n_dom_landunits to the bottom of subr surfrd_special and pass back&forth the wt_lunit instead of the five pct* variables.

Reply:
Awesome